controllers/velero_test.go

```diff
@@ -1285,10 +1489,16 @@ func TestDPAReconciler_buildVeleroDeployment(t *testing.T) {
 			r := DPAReconciler{
 				Client: fakeClient,
 			}
-			if err := r.buildVeleroDeployment(tt.veleroDeployment, tt.dpa); (err != nil) != tt.wantErr {
+			err = nil
```

Comment:
Why do we need such a pattern ? Isn't the existing testing one good enough ?

Reply:
Previous pattern, err initialized err `:=` is only accessible in the if statement that initialized that err.

If you revert this piece to previous pattern `err` will be nil in line [1496](https://github.com/openshift/oadp-operator/pull/588/files#diff-e67036eb3425424478a1a6ac6b7f49162a22db305f67038c0d3e1f8e4c429211R1496) even when `r.buildVeleroDeployment` returns an error.

Instead of reinitializing inside of if statement which encapsulates the err to that if statement, we reuse err made earlier outside if statement.

Reply:
so that err can be queried in line [1496](https://github.com/openshift/oadp-operator/pull/588/files#diff-e67036eb3425424478a1a6ac6b7f49162a22db305f67038c0d3e1f8e4c429211R1496)

Reply:
actually a better pattern might be to put dependency of this err in here..

Reply:
Refactored in https://github.com/openshift/oadp-operator/pull/588/commits/6d306e5803a9a8dafa054d1d8eef28bcc5939d7e